lib/reporter.js

```diff
@@ -336,6 +336,7 @@ class Reporter {
 
     // Finishes launch
     onDone (err, result) {
+        const checkedFailures = result && result.run && result.run.failures && result.run.failures.length;
```

Comment:
In my opinion, it is better to create function, and name it like hasFailuares or something like this. It will be looks more clear.
And it looks like we do not need it at all.
Maybe better to handle it like this?
status: status || (result.run.failures.length ? TestStatus.FAILED : TestStatus.PASSED)
Because already existing logic works like this :
If we have status, we do not have any other results, if not, check result.run.failures

Reply:
If we do like that, tests will failed

Reply:
you can try to write something like this `(result.run.failures || []).length` to avoid errors since result.run.failures worked fine before